packages/headless/src/features/instant-results/instant-result-analytics-actions.ts

```diff
@@ -0,0 +1,33 @@
+import {InstantResultsAnalyticsProvider} from '../../api/analytics/instant-result-analytics';
+import {Result} from '../../api/search/search/result';
+import {
+  partialDocumentInformation,
+  documentIdentifier,
+  validateResultPayload,
+  makeAnalyticsAction,
+  AnalyticsType,
+  ClickAction,
+  SearchAction,
+} from '../analytics/analytics-utils';
+
+export const logInstantResultOpen = (result: Result): ClickAction =>
+  makeAnalyticsAction(
+    'analytics/instantResult/open',
+    AnalyticsType.Click,
+    (client, state) => {
+      validateResultPayload(result);
+      return client.makeDocumentOpen(
+        partialDocumentInformation(result, state),
+        documentIdentifier(result)
+      );
+    },
+    (getState) => new InstantResultsAnalyticsProvider(getState)
+  );
+
+export const logInstantResultsSearch = (): SearchAction =>
+  makeAnalyticsAction(
+    'analytics/instantResult/searchboxAsYouType',
+    AnalyticsType.Search,
+    (client) => client.makeSearchboxAsYouType(),
+    (getState) => new InstantResultsAnalyticsProvider(getState)
```

Comment:
Now that both events have this provider, no more issues, it will get the Search ID from the "active" instant result list.